File: services/tradelog/tradelog-api/src/main/java/com/example/tradelog/api/converter/protobuf/TransactionTypeConverter.java
package com.example.tradelog.api.converter.protobuf;


import java.util.function.Function;

public class TransactionTypeConverter {

    public static Function<com.example.tradelog.api.spec.model.TransactionType, com.example.tradelog.api.core.model.TransactionType> toModel = protoModel -> {
        return com.example.tradelog.api.core.model.TransactionType.Companion.lookup(protoModel.name());
    };

    public static Function<com.example.tradelog.api.core.model.TransactionType, com.example.tradelog.api.spec.model.TransactionType> toProto = apiModel -> {
        return com.example.tradelog.api.spec.model.TransactionType.valueOf(apiModel.name());
    };
}

File: services/gateway/gateway-api/src/main/kotlin/com/example/gateway/api/rest/converter/TransactionTypeConverter.kt
package com.example.gateway.api.rest.converter

import com.example.gateway.api.core.model.TransactionType
import com.example.gateway.api.spec.model.ShareJournalDto as GWShareJournalDto
import com.example.tradelog.api.spec.model.TransactionDto as TLTransactionDto

class TransactionTypeConverter {

    companion object {
        fun toModel(dto: TLTransactionDto.TransactionType): TransactionType {
            return TransactionType.valueOf(dto.name)
        }

        fun toDto(model: TransactionType): GWShareJournalDto.TransactionType {
            return GWShareJournalDto.TransactionType.valueOf(model.name)
        }
    }
}